administration/views.py

```diff
@@ -1,49 +1,97 @@
-from rest_framework.permissions import IsAuthenticated, IsAdminUser
+from rest_framework.permissions import (
+    IsAuthenticated,
+    IsAdminUser,
+    BasePermission,
+)
 from rest_framework.generics import (
     ListAPIView,
     ListCreateAPIView,
     RetrieveUpdateDestroyAPIView,
+    RetrieveAPIView,
 )
 
 from administration.serializers import (
-    AdminUserSerializer,
     AdminCompanyListSerializer,
     AdminCompanyDetailSerializer,
+    AdminUserListSerializer,
+    AdminUserDetailSerializer,
+    AdminUserStatusSerializer,
 )
 from administration.pagination import ListPagination
 from authentication.models import CustomUser
 from profiles.models import Profile
+from rest_framework.response import Response
+from rest_framework import status
+
+
+class IsStafUser(BasePermission):
+    """
+    Custom is staff permission.
+    """
+
+    def has_permission(self, request, view):
+        return request.user.is_staff
 
 
 class UsersListView(ListAPIView):
     """
     List of users.
     """
 
-    queryset = CustomUser.objects.filter(is_superuser=False).order_by("id")
-    serializer_class = AdminUserSerializer
-    permission_classes = [IsAuthenticated, IsAdminUser]
+    permission_classes = [IsAuthenticated, IsStafUser, IsAdminUser]
     pagination_class = ListPagination
+    serializer_class = AdminUserListSerializer
+    queryset = CustomUser.objects.all().order_by("id")
 
 
 class UserDetailView(RetrieveUpdateDestroyAPIView):
     """
     Retrieve, update or delete a user.
     """
 
-    queryset = CustomUser.objects.filter(is_superuser=False).order_by("id")
-    permission_classes = [IsAuthenticated, IsAdminUser]
-    serializer_class = AdminUserSerializer
+    permission_classes = [IsAuthenticated, IsStafUser, IsAdminUser]
```

Comment:
Any reason to have here _both_ `IsStaffUser` and `IsAdminUser`? Doesn't `IsAdminUser` include the logic of `IsStaffUser`?

Reply:
= > [IsAuthenticated, IsStaffUser]